src/macports1.0/macports.tcl

```diff
@@ -775,11 +882,29 @@ proc mportinit {{up_ui_options {}} {up_options {}} {up_variations {}}} {
         if {[file exists $file]} {
             set portconf $file
             set fd [open $file r]
+
+            # List of multivalue options that did already occur in this
+            # configuration file. This enables the mechanism that allows
+            # specifying a line multiple times to specify a list, but still
+            # allows starting from scratch in the next configuration file
+            array set options_in_file {}
             while {[gets $fd line] >= 0} {
                 if {[regexp $conf_option_re $line match option ignore val] == 1} {
                     if {$option in $bootstrap_options} {
+                        # standard single-value options
                         set macports::$option [string trim $val]
                         global macports::$option
+                    } elseif {$option in $bootstrap_multivalue_options} {
```

Comment:
I'm not terribly happy about having two different behaviours for options in the config file. Is it just to avoid long lines?

Reply:
Yes. Do you have a better suggestion on how to achieve this?